Libraries/Opc.Ua.Configuration/ApplicationInstance.cs

```diff
@@ -433,12 +449,27 @@ public async Task DeleteApplicationInstanceCertificate(CancellationToken ct = de
         /// <param name="silent">if set to <c>true</c> no dialogs will be displayed.</param>
         /// <param name="minimumKeySize">Minimum size of the key.</param>
         /// <param name="lifeTimeInMonths">The lifetime in months.</param>
-        /// <param name="ct">The cancellation token.</param>
+        /// <param name="ct"></param>
+        [Obsolete("This method is obsolete since an application now supports different minKey sizes depending on certificate type")]
         public async Task<bool> CheckApplicationInstanceCertificate(
             bool silent,
             ushort minimumKeySize,
             ushort lifeTimeInMonths,
             CancellationToken ct = default)
+        {
```

Comment:
the non obsolete functions should have the Async suffix, these can remain until removed.